packages/api2/src/users/users.service.ts

```diff
@@ -146,4 +147,30 @@ export class UsersService {
     const createdUser = new this.userModel(userDto);
     return createdUser.save();
   }
+
+  /**
+   * Generate username from user account name
+   * If username is already taken than create one with discriminator
+   *
+   * @param userAccount
+   * @returns {Promise<string>}
+   */
+  generateUserNameFromAccount = async (
```

Comment:
Should the service class include methods that do not modify data? Or should those methods be in a utility class? But in that case we would like to make the utility class a provider. UsersUtilsProvider, hrm. Doesn't feel a 100% either. What do you think?